pages/index.js

```diff
@@ -1,19 +1,28 @@
 import Days from "../src/components/Days";
 import Link from "next/link";
 import useStore from "../src/store";
-import { useState, useEffect, useCallback } from "react";
-import { sessions } from "../src/sessionsDb";
+import { useState } from "react";
+import useSWR from "swr";
+import EventDistances from "../src/components/EventDistances";
 
 export default function HomePage() {
   const { days, toggleDay } = useStore();
   const addedDays = days && days.filter((day) => day.added);
   const [selectedType, setSelectedType] = useState("short");
 
-  const [isLoading, setIsLoading] = useState(true);
+  const { data: sessions, error } = useSWR("/api/sessions", async (url) => {
+    try {
+      const res = await fetch(url);
+      const data = await res.json();
+      return data;
+    } catch (error) {
+      throw new Error("Error loading sessions");
+    }
+  });
```

Comment:
you could declare a fetcher, instead of doing the callback directly within the arguments.